src/zsim.cpp

```diff
@@ -960,9 +1024,12 @@ VOID ContextChange(THREADID tid, CONTEXT_CHANGE_REASON reason, const CONTEXT* fr
             break;
         case CONTEXT_CHANGE_REASON_SIGNAL:
             reasonStr = "SIGNAL";
+            signalStackDepth[tid]++;
             break;
         case CONTEXT_CHANGE_REASON_SIGRETURN:
             reasonStr = "SIGRETURN";
+            assert(signalStackDepth[tid] > 0);
+            signalStackDepth[tid]--;
             break;
         case CONTEXT_CHANGE_REASON_APC:
             reasonStr = "APC";
```

Comment:
Just a question: do the other context switch reasons need to be similarly handled?